app/src/main/java/com/example/caregiver/EditTask.java

```diff
@@ -0,0 +1,167 @@
+        Intent intent = getIntent();
+        currTask = intent.getParcelableExtra("currTask");
+        String[] caregiveeRooms = intent.getStringArrayExtra("rooms");
+        String caregiveeName = intent.getStringExtra("caregiveeName");
+
+        /* TODO better error handling */
+        // if task is not found, do not display
+        if(currTask == null) return;
+
+        // get all field/spinners
+        taskNameField = findViewById(R.id.taskName);
+        taskNotesField = findViewById(R.id.taskNotes);
+        roomSpinner = findViewById(R.id.taskRoom);
+        caregiveeField = findViewById(R.id.taskCaregivee);
+
+        // populate all fields with task info & create spinner
+        taskNameField.setText(currTask.taskName, TextView.BufferType.EDITABLE);
+        taskNotesField.setText(currTask.taskNote, TextView.BufferType.EDITABLE);
+        caregiveeField.setText(caregiveeName, TextView.BufferType.NORMAL);
+        createSpinner(caregiveeRooms, currTask.room);
+
+        Log.i("INFO", "EditTask:onCreate displayed all task details successfully.");
+
+        /* TODO handle task removal */
+    }
+
+    /**
+     * Initializes room spinner, creates adapter with list of all caregivee rooms
+     * @param allCaregiveeRooms, list of caregivee rooms
+     * @param currRoom, room of current task
+     */
+    protected void createSpinner(String[] allCaregiveeRooms, String currRoom){
+        // set curr room to first entry
+        for(int i = 0; i < allCaregiveeRooms.length; i++){
+            // swap first with curr room
+            if(allCaregiveeRooms[i].equals(currRoom)){
+                String temp = allCaregiveeRooms[i];
+                allCaregiveeRooms[i] = allCaregiveeRooms[0];
+                allCaregiveeRooms[0] = temp;
+                break;
+            }
+        }
+        // Render list on the caregivee spinner
+        ArrayAdapter<String> adapter = new ArrayAdapter<String> (
+                this, android.R.layout.simple_spinner_item,  allCaregiveeRooms);
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+       roomSpinner.setAdapter(adapter);
+    }
+
+    /**
+     * Cancel Button onClick function - navigates back to dashboard
+     * @param view
+     */
+    public void NavigateToDashboard(View view){
+        // navigate to dashboard after update
+        Intent intent = new Intent(this, Dashboard.class);
+        startActivity(intent);
+    }
+
+    /**
+     * Updates task in firebase, triggered when user clicks 'Update'
+     * If room is updated, the previous task is removed
+     * The updated task is written to the DB
+     * @param view
+     */
+    public void UpdateTask(View view) {
+        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
+        String path;
+        String updatedRoom = roomSpinner.getSelectedItem().toString();
+
+        // if room changed, remove previous task
```

Comment:
Small nit. Just keep one comment that says "if room changed, remove the old task from its room." or combine the two comments.

Reply:
Good catch lol that was v redundent